### PR TITLE
DRF Logout

### PR DESCRIPTION
**Pull request by M-Muaz-Shoukat**

### CODE REVIEW

File: todo_list/settings.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
     'django_celery_results',
     'todo_list_app',
     'django_celery_beat',
+    'rest_framework_simplejwt.token_blacklist',
```
Comment (m-shayan-ahmad) [inline todo_list/settings.py:45]: What is this app used for?

File: todo_list_app/serializers.py [hunk 2]
```diff
@@ -37,5 +39,20 @@ class Meta:
...
+        'bad_token': 'Token is invalid or has expired!',
+    }
+
+    def validate(self, attrs):
+        self.token = attrs.get('refresh_token')
+        return attrs
+
+    def save(self, **kwargs):
+        try:
+            RefreshToken(self.token).blacklist()
```
Comment (m-shayan-ahmad) [inline todo_list_app/serializers.py:55]: What happens to a blacklisted token?

**Review by m-shayan-ahmad — APPROVED** [review]

File: todo_list_app/serializers.py (hunk 2 above)
Comment (M-Muaz-Shoukat) [reply todo_list_app/serializers.py:55]: Blacklist token and outstanding token remains in the list until they get expired and we can delete them by periodically running delete command on server

File: todo_list/settings.py (hunk 1 above)
Comment (M-Muaz-Shoukat) [reply todo_list/settings.py:45]: This app manages two list outstanding token and blacklisted token initially it add all tokens to outstanding list and when the user logout it mark that token in blacklisted token list corresponding to outstanding token list

Comment (m-shayan-ahmad) [reply todo_list/settings.py:45]: Can we see the blacklisted tokens?

File: todo_list_app/serializers.py (hunk 2 above)
Comment (m-shayan-ahmad) [reply todo_list_app/serializers.py:55]: Can we create a job to delete the blacklisted tokens after some time?

**Comment by m-shayan-ahmad** [comment]
We will be implementing a feature to delete blacklisted tokens automatically after a set period of time.

**Pull request merged**